### ISSUE TITLE
Describing data types for models

### ISSUE THREAD

**Issue by IgoP02** [ad-on-is/adonis-autoswagger#51]
Is it possible to describe data types in schemas using keywords? After reading the guide I haven't been able to think of a way to do it. 

I've also tried directly modifying the generated swagger.yml file and adding some keywords:

```
    Brand:
      type: "object"
      properties:
        id:
          type: "number"
          example: 164
        name:
          type: "string"
          example: "John Doe"
          minLength: 1
          maxLength: 100
        created_by:
          type: "number"
          example: 638
```
But even then they won't show up the in schema section on the docs:

![image](https://github.com/ad-on-is/adonis-autoswagger/assets/40871105/86f6f214-a4fe-4a35-a478-d5d4257a45da)

**Comment by ad-on-is** [comment]
I'm afraid, I don't fully understand your question?

Do you want to inject a schema that has no Model associated to it? Or do you want to add your own examples for specific fields?

**Comment by IgoP02** [comment]
I want to add keywords such as maxLength, minLength, etc. to my schema descriptions. Like this:

![image](https://github.com/ad-on-is/adonis-autoswagger/assets/40871105/58afd043-c2c5-43fd-a52b-af2cd1f7235d)

Done with the online Swagger editor.

**Comment by ad-on-is** [comment]
> I want to add keywords such as maxLength, minLength, etc. to my schema descriptions. Like this:
> 
> ![image](https://private-user-images.githubusercontent.com/40871105/306740955-58afd043-c2c5-43fd-a52b-af2cd1f7235d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg1NDIxMjAsIm5iZiI6MTcwODU0MTgyMCwicGF0aCI6Ii80MDg3MTEwNS8zMDY3NDA5NTUtNThhZmQwNDMtYzJjNS00M2ZkLWE1MmItYWYyY2QxZjcyMzVkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjIxVDE4NTcwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTViZDM3ZDUxMzRkMTEwNmE4Nzk2MzE5MjE5ZDY2MWE1NGE5ZmNkZjI4ZDhjNGY5NDFjY2ZhYzg5OWZlNTI3NWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.zA8-mHANYvRJSNgwurwadF9qNRb8o7Ra2IYBoSNJlI4)
> 
> Done with the online Swagger editor.

Ohhh I see.
Sorry, this is not implemented. You can submit a PR if you're willing to? The best place for this would be to do it the same way `@example` does it, in the model-file itself.

**Comment by ad-on-is** [comment]
I see there are different keywords for each data type, so this might take a while to implement and test thoroughly.

**Comment by IgoP02** [comment]
Not sure about the implications of using specific keywords, but it does seem to display any property you pass it.
![image](https://github.com/ad-on-is/adonis-autoswagger/assets/40871105/aee2a6b9-17c7-4a78-b4ef-811681309724)
Here `min` and `max `are not keywords provided in the swagger specification. Maybe just using a user provided name and value could work to some extent.

I'm not sure if I have the time right now, but if I can work it out soon, I'll create a PR.

**Comment by ad-on-is** [comment]
Oh that's nice... then something like @keywords({minvalue":10, maxvalue:20})

**Comment by IgoP02** [comment]
That seems to be a good way to do it. It'd be a matter of testing to see if there are no major issues with using additional keywords, though in that case, a valid keyword list could be provided so users can use the appropriate ones for each data type.

**Comment by ad-on-is** [comment]
> That seems to be a good way to do it. It'd be a matter of testing to see if there are no major issues with using additional keywords, though in that case, a valid keyword list could be provided so users can use the appropriate ones for each data type.

I think `{0xmick3y: M0u5e}` should be sufficiant to test on all types... if that works anything will work 😂

**Comment by IgoP02** [comment]
If that works, then anything will :sweat_smile:

**Comment by ad-on-is** [comment]
Fixed in 3.25
